lib/task_listener.js

```diff
@@ -164,7 +168,8 @@ export default class TaskListener extends EventEmitter {
     if (exceedsThreshold) return;
 
     let claims = await this.taskQueue.claimWork(availableCapacity);
-    claims.forEach(this.runTask.bind(this));
+    let tasksets = await Promise.all(claims.map(this.coalesceClaim.bind(this)));
+    await Promise.all(tasksets.map(this.runTaskset.bind(this)));
```

Comment:
Oh, good point.  The old code leaves un-handled promises, so I was trying to fix that -- is there a good quick way to say "if this promise fails, log it, otherwise I don't care"?